rfq/oracle.go

```diff
@@ -60,6 +62,43 @@ type OracleBidResponse struct {
 	Err *OracleError
 }
 
+// OracleAddr is a type alias for a URL type that represents a price oracle
+// service address.
+type OracleAddr = url.URL
+
+const (
+	// RfqRpcOracleAddrScheme is the URL address scheme used by an RPC price
+	// oracle service.
+	RfqRpcOracleAddrScheme string = "rfqrpc"
+)
+
+// ParsePriceOracleAddress parses a price oracle service address string and
+// returns a URL type instance.
+func ParsePriceOracleAddress(addrStr string) (fn.Option[OracleAddr], error) {
+	addrNone := fn.None[OracleAddr]()
+
+	// Basic sanity check to ensure the address is not empty.
+	if addrStr == "" {
```

Comment:
I think we should restrict the use of `fn.Option` to places where returning a nil error and `fn.None()` is actually possible. Since here we always either return an error or a valid address, we shouldn't use options.

Reply:
Does that mean that the return value becomes a pointer I suppose?